src/Utility/Composer.php

```diff
@@ -133,6 +151,109 @@ protected static function getUpdateConfig($directory, $repository)
+            return $config;
+        }
+
+        $requirements = $composerData['require'];
+
+        foreach ($requirements as $package => $version) {
+            if ($package === 'php') {
+                if (isset($composerData['config']['platform']['php'])) {
+                    continue;
+                }
+
+                $versions = static::parseVersions($version);
+
+                if (count($versions)) {
+                    $config['platform.php'] = [$versions[0]];
+                }
+            } elseif (strpos($package, '/') === false && substr($package, 0, 4) === 'ext-') {
+                if (isset($composerData['config']['platform'][$package])) {
+                    continue;
+                }
+
+                $packageKey = sprintf('platform.%s', $package);
+
+                if ($version === '*') {
+                    $config[$packageKey] = ['1'];
+                } else {
+                    $versions = static::parseVersions($version);
+
+                    if (count($versions)) {
+                        $config[$packageKey] = [$versions[0]];
+                    }
+                }
+            }
+        }
+
+        return $config;
+    }
+
+    /**
+     * Read the allowed versions defined in the composer format and return them ordered
+     *
+     * @param String $versionDefinition version definition
+     * @return array ordered list of versions as strings
+     *
+     * @link https://getcomposer.org/doc/articles/versions.md Composer version format documentation
+     *
+     */
+    private static function parseVersions($versionDefinition)
+    {
+        static $regex = '/
+            (?<!(?:<|>|\.|\d))  # Does not start with "<", ">", "." or a number
+                                # that means it may be ">=7.1", "<=7.1", "7.1" but cannot be ">7.1" nor "<7.1"
+                                # We also eliminate partial matches ignoring stuff starting with
+                                # a number or a dot ".", so that ">7.1.25" does not become "1.25" nor "5"
+
+            (?<version>\d+(?:\.\d+(?:\.\d+)?)?)
+        /x';
+
+        if (preg_match_all($regex, $versionDefinition, $matches)
+            && isset($matches['version'])
+            && count($matches['version'])
+        ) {
+            $versions = $matches['version'];
+
+            usort($versions, static function ($a, $b) {
+                $af = floatval($a);
+                $bf = floatval($b);
+
+                if ($af < $bf) {
+                    return -1;
+                } elseif ($af > $bf) {
+                    return 1;
+                } else {
+                    if ($a < $b) {
+                        return -1;
+                    } elseif ($a > $b) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                }
+            });
+
+            return $versions;
+        } else {
+            return [];
+        }
```

Comment:
I'm pretty sure there are utility methods you can use to sort versions/constraints - have you had a look at https://github.com/composer/semver?

Reply:
yes, I was thinking about that one.
Unfortunately, we cannot reuse much from there in regards to parsing versions (at least without making things more complicated than it needs to be), and adding this 3rd party dependency for only comparing the versions seems to be an overkill to me...
On the other hand, we have this bit isolated in a private function, so it cannot spread across the app. If we need the semver functionality later some day, we could add the dependency and refactor this function as well.

Reply:
Sweet!